src/notebooks/controllers/controllerRegistration.ts

```diff
@@ -195,6 +195,9 @@ export class ControllerRegistration implements IControllerRegistration {
 
         if (this.inKernelExperiment) {
             return userFiltered || connectionTypeFiltered || urlFiltered;
+        } else if (this.configuration.getSettings().kernelPickerType === 'Insiders') {
```

Comment:
Minor change to controllerRegistration. Without this urls that are not the current url get filtered out.